File: 1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.go
func checkIfExist(arr []int) bool {
    visited := make(map[int]bool)
    for _, v := range arr {
        if visited[2*v] {
            return true
        }
        if v%2==0 && visited[v/2] {
            return true
        }
        visited[v] = true
    }
    return false
}

File: 1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.rs
use std::collections::HashSet;

impl Solution {
    pub fn check_if_exist(arr: Vec<i32>) -> bool {
        let mut nums : HashSet<i32> = HashSet::new();

        for num in arr {
            if nums.contains(&num) {
                return true;
            }
            if num % 2 == 0 {
                nums.insert(num / 2);
            }
            nums.insert(num * 2);
        }

        false
    }
}